src/Rules/Variables/UnsetRule.php

```diff
@@ -103,6 +103,19 @@ private function canBeUnset(Node $node, Scope $scope): ?IdentifierRuleError
 					->identifier($propertyReflection->isReadOnly() ? 'unset.readOnlyProperty' : 'unset.readOnlyPropertyByPhpDoc')
 					->build();
 			}
+
+			if ($propertyReflection->getNativeReflection()->getHooks() !== []) {
```

Comment:
I'd rather be if you asked this on `$propertyReflection` directly. You can add a method `isHooked(): bool` which does `$this->getHook !== null || $this->setHook !== null`.